web/crux-ui/src/components/products/versions/images/edit-image-json.tsx

```diff
@@ -82,61 +137,168 @@ const mergeKeyValuesWithJson = (items: UniqueKeyValue[], json: Record<string, st
+
+            return {
+              ...vit,
+              id,
+            } as InitContainerVolumeLink
+          }),
+        } as InitContainer
+      }
+
+      return {
+        ...it,
+        id: uuid(),
+        command: it.command?.map(cit => ({ id: uuid(), key: cit })),
+        args: it.args ? it.args?.map(ait => ({ id: uuid(), key: ait })) : [],
+        environments: Object.keys(it.environments ?? {}).map(eit => ({
+          key: eit,
+          value: it.environments[eit],
+          id: uuid(),
+        })),
+        volumes: it.volumes?.map(vit => ({ ...vit, id: uuid() })),
+      } as InitContainer
+    }),
+    ports: json.ports?.map(it => ({
+      ...it,
+      id: uuid(),
+    })),
+    portRanges: json.portRanges?.map(it => ({
+      ...it,
+      id: uuid(),
+    })),
+  } as Partial<ContainerConfig>
+
+  if ((json as JsonContainerConfig).secrets) {
+    config.secrets = (json as JsonContainerConfig).secrets.map(it => {
+      const prev = serialized.secrets?.map(sit => sit.key).indexOf(it)
+
+      return {
+        id: prev !== -1 ? serialized.secrets[prev].id : uuid(),
+        key: it,
+        value: '',
+      }
+    })
+  }
+
+  return config
 }
 
 interface EditImageJsonProps {
   disabled?: boolean
   className?: string
-  config: ContainerConfig | InstanceContainerConfig
+  config: ContainerConfig
   editorOptions: EditorStateOptions
-  onPatch: (config: Partial<ContainerConfig | InstanceContainerConfig>) => void
+  onPatch: (config: Partial<ContainerConfig>) => void
   onParseError?: (err: Error) => void
+  instanceEditor?: boolean
 }
 
 const EDITOR_ID = 'json-config'
-const JSON_EDITOR_COMPARATOR = (one: CompleteContainerConfig, other: CompleteContainerConfig): boolean =>
+const JSON_EDITOR_COMPARATOR = (one: JsonConfig, other: JsonConfig): boolean =>
   JSON.stringify(one) === JSON.stringify(other)
 
 const EditImageJson = (props: EditImageJsonProps) => {
-  const { disabled, editorOptions, className, config, onPatch, onParseError: propOnParseError } = props
+  const { disabled, editorOptions, className, config, onPatch, onParseError: propOnParseError, instanceEditor } = props
 
   const throttle = useThrottling(IMAGE_WS_REQUEST_DELAY)
 
-  const onMergeValues = (_: CompleteContainerConfig, local: CompleteContainerConfig): CompleteContainerConfig => {
-    onPatch({
-      config: local,
-      name: local.name,
-      environment: mergeKeyValuesWithJson(config?.environment ?? [], local?.environment),
-      capabilities: mergeKeyValuesWithJson(config?.capabilities ?? [], local?.capabilities),
-    })
+  const onMergeValues = (_: JsonConfig, local: JsonConfig): JsonConfig => {
+    onPatch(merge(config, local))
 
     return local
   }
 
   const [editorState, editorActions] = useMultiInputState({
     id: EDITOR_ID,
-    value: imageConfigToCompleteContainerConfig(null, config),
+    value: instanceEditor
+      ? imageConfigToInstanceJsonConfig(null, config)
```

Comment:
`imageConfigToInstanceJsonConfig` rename to `imageConfigToJsonInstanceConfig` like above function.